packages/modelviewer.dev/src/docs-and-examples/create-html.ts

```diff
@@ -0,0 +1,276 @@
+  entryLink.classList.add('darken');
+  entryLink.innerText = entry.name;
+  entrySidebar.appendChild(entryLink);
+  return entrySidebar
+}
+
+function createSubcategorySidebar(
+    key: string, lowerCaseTitle: string, category: Category) {
+  let lowerCaseKey = getLowerCaseKey(key);
+
+  // Create subcategory container
+  let subcategorySidebarContainer = document.createElement('div');
+  subcategorySidebarContainer.classList.add('subCategory');
+
+  // Create subcategory header
+  let subcategoryHeader = document.createElement('H4');
+  subcategoryHeader.id = lowerCaseTitle.concat('-', lowerCaseKey, '-sidebar');
+  subcategoryHeader.classList.add('subcategory-header');
+
+  // Create subcategory link
+  let subcategoryLink = document.createElement('a');
+  subcategoryLink.setAttribute(
+      'href', '#'.concat(lowerCaseTitle, '-', lowerCaseKey));
+  subcategoryLink.classList.add('darken');
+  subcategoryLink.innerText = key;
+
+  subcategoryHeader.appendChild(subcategoryLink);
+  subcategorySidebarContainer.append(subcategoryHeader);
+
+  const innerKey = key as keyof typeof CategoryConstant;  // hack to use later
+  (<Entry[]>category[innerKey]).forEach(function(entry: Entry) {
+    let entrySidebar = createEntrySidebar(lowerCaseTitle, lowerCaseKey, entry);
+    subcategorySidebarContainer.append(entrySidebar);
+  });
+  return subcategorySidebarContainer
+}
+
+function createSidebar(category: Category, docsOrExamples: string) {
+  let container = document.getElementById('sidebar-category-container');
+  let lowerCaseTitle = category.Title.toLowerCase();
+
+  let categoryContainer = document.createElement('div');
+  categoryContainer.classList.add('category');
+
+  let categoryHeader = document.createElement('H3');
+  categoryHeader.id = lowerCaseTitle.concat('-sidebar');
+
+  let categoryLink = document.createElement('a');
+  categoryLink.setAttribute('href', '#'.concat(lowerCaseTitle))
+  categoryLink.classList.add('darken');
+  categoryLink.innerText = category.Title;
+
+  categoryHeader.appendChild(categoryLink);
+  categoryContainer.append(categoryHeader);
+
+  if (docsOrExamples === 'examples') {
+    container!.appendChild(categoryContainer);
+    return;
+  }
+
+  Object.keys(category).forEach((key) => {
+    if (key !== 'Title') {
+      let subcategorySidebarContainer =
+          createSubcategorySidebar(key, lowerCaseTitle, category);
+      categoryContainer.appendChild(subcategorySidebarContainer);
+    }
+  });
+  container!.appendChild(categoryContainer);
+}
+
+function createTitle(header: string, docsOrExamples: string) {
+  let lowerCaseHeader = header.toLowerCase();
+  let element: HTMLElement;
+  let para = document.createElement('div');
+  para.classList.add('header');
+  let node = document.createElement('H1');
+  node.id = lowerCaseHeader;
+  node.innerText = header;
+  para.appendChild(node);
+  if (docsOrExamples === 'docs') {
+    element = document.getElementById(lowerCaseHeader.concat('-docs'))!;
```

Comment:
Try replacing this whole if statement block with something like
```
const name = docsOrExamples === 'docs' ? '-docs' : '-examples-header';
document.getElementById(lowerCaseHeader.concat(name))!.appendChild(para);
```

Reply:
done